components/ecwid/sources/get-paid-orders/get-paid-orders.mjs

```diff
@@ -0,0 +1,50 @@
+import ecwid from "../../ecwid.app.mjs";
+
+export default {
+  name: "Ecwid Paid Orders",
+  version: "0.0.1",
+  key: "ecwid-paid-orders",
+  description: "Search for new orders which are PAID and AWAITING_PROCESSING. Emits events for each order and" +
+      " sets order fulfilment status to PROCESSING",
+  props: {
+    timer: {
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: 60 * 15,
+      },
+    },
+    history: {
+      label: "Max Days Back",
+      description: "Max number of days to get the orders from",
+      type: "integer",
+      default: 30,
+    },
+    setFulfilmentStatus: {
+      label: "Set fulfilment status to PROCESSING (Preferred)",
+      description: "Upon receiving the order, fulfilment status will be set to PROCESSING so that " +
+          "the same order is not fetched again. If unchecked, it needs to be handled in the workflow",
+      type: "boolean",
+      default: true,
+    },
+    ecwid,
+  },
+  type: "source",
+  methods: {},
```

Comment:
Try to use deploy hook to best experience.

https://pipedream.com/docs/components/api/#deploy

```suggestion
  methods: {},
  hooks: {
    deploy() {
      // deploy code here
    },
  },
```

Reply:
At present, we don't need a webhook. Ecwid uses a centralised webhook for entire app and registration of webhook is manual (By requesting via customer service). It does not really fit with creating webhook per deployed components.

Reply:
Hi @trendoscope-algorithms, @lcaresia, as I understand, historical records are already emitted here. So I think that we do not need `deploy` hook. However, if I get it correctly, whenever the timer ticks this one is going to emit all related orders again. We can avoid this by setting a db record and check it before making the request. Please check the db docs [here](https://pipedream.com/docs/components/api/#db) and an example [here](https://github.com/PipedreamHQ/pipedream/blob/f31508b6fa735a0b3ebafccbd84e5f86028ec6ef/components/pinterest/sources/common.mjs#L16). As I see [here](https://api-docs.ecwid.com/reference/search-orders#orderentry), there are `createDate` and `updateDate` props in `OrderEntry`, I think these can help while getting new or updated orders.

Reply:
Hi @feyzullah , That is been handled by setting the order status to processing. When we are fetching the new orders, we only fetch those which are in the status awaiting processing.

User also has option to set status to processing if they are handling that in the workflow. Hence, this is given an option which is selected by default.

Reply:
Hi @trendoscope-algorithms, As I understand, that logic is implemented on Ecwid side as only items with awaiting status are fetched and those items are the new ones. This one looks good to me but Jorge is on call for PR reviews for this week. Let's request his review.